### PR TITLE
feat(compute): compute_ip_address_promote_ephemeral sample

### PR DESCRIPTION
**Pull request by irudykss**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [ ] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [ ] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Comment by gotochkin** [comment]
Reviewed as part of DevRel Learning Days - Samples training

File: compute/address/address_test.go [hunk 1]
```diff
@@ -628,3 +628,98 @@ func TestAssignStaticAddressToExistingVM(t *testing.T) {
...
+	addresses, err = listIPAddresses(ctx, tc.ProjectID, region)
+	if err != nil {
+		t.Errorf("listIPAddresses got err: %v", err)
+	}
+
+	for _, address := range addresses {
+		if strings.Contains(address, addressName) {
+			return
+		}
+	}
```
Comment (m-strzelczyk) [inline compute/address/address_test.go:723]: Shouldn't you release the IP address somewhere here? Or even earlier with the whole `defer` thing? The whole point of static IP address it that it is not released after a VM is deleted. Check your test project, you probably have a bunch of unreleased IPs, generating cost.

**Review by m-strzelczyk — CHANGES_REQUESTED** [review]

File: compute/address/address_test.go (hunk 1 above)
Comment (BigBlackWolf) [reply compute/address/address_test.go:729]: Good catch! Thanks for pointing

**Review by muncus — APPROVED** [review]

**Comment by muncus** [comment]
the test failures are for N2 CPUs in europe, which are not used by this PR, so they're unrelated.

**Pull request merged**